tests/wgsl/texture.slang

```diff
@@ -0,0 +1,657 @@
+Texture2DArray<int16_t4> t2DArray_i16;
+//TEST_INPUT: TextureCube(size=4, content = zero, arrayLength=2):name tCubeArray_i16
+TextureCubeArray<int16_t4> tCubeArray_i16;
+
+//TEST_INPUT: Texture1D(size=4, content = zero):name t1D_u16
+Texture1D<uint16_t4> t1D_u16;
+//TEST_INPUT: Texture2D(size=4, content = zero):name t2D_u16
+Texture2D<uint16_t4> t2D_u16;
+//TEST_INPUT: Texture3D(size=4, content = zero):name t3D_u16
+Texture3D<uint16_t4> t3D_u16;
+//TEST_INPUT: TextureCube(size=4, content = zero):name tCube_u16
+TextureCube<uint16_t4> tCube_u16;
+
+//TEST_INPUT: Texture1D(size=4, content = zero, arrayLength=2):name t1DArray_u16
+Texture1DArray<uint16_t4> t1DArray_u16;
+//TEST_INPUT: Texture2D(size=4, content = zero, arrayLength=2):name t2DArray_u16
+Texture2DArray<uint16_t4> t2DArray_u16;
+//TEST_INPUT: TextureCube(size=4, content = zero, arrayLength=2):name tCubeArray_u16
+TextureCubeArray<uint16_t4> tCubeArray_u16;
+
+// Metal doc says "For depth texture types, T must be float."
+__generic<T : __BuiltinType, let sampleCount:int=0, let format:int=0>
+typealias depth2d = __TextureImpl<
+    T,
+    __Shape2D,
+    0, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    0, // isCombined
+    format
+>;
+
+__generic<T : __BuiltinType, let sampleCount:int=0, let format:int=0>
+typealias depth2d_array = __TextureImpl<
+    T,
+    __Shape2D,
+    1, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    0, // isCombined
+    format
+>;
+
+__generic<T : __BuiltinType, let sampleCount:int=0, let format:int=0>
+typealias depthcube = __TextureImpl<
+    T,
+    __ShapeCube,
+    0, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    0, // isCombined
+    format
+>;
+
+__generic<T : __BuiltinType, let sampleCount:int=0, let format:int=0>
+typealias depthcube_array = __TextureImpl<
+    T,
+    __ShapeCube,
+    1, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    0, // isCombined
+    format
+>;
+
+//TEST_INPUT: Texture2D(size=4, content = zero):name d2D
+depth2d<float> d2D;
+//TEST_INPUT: TextureCube(size=4, content = zero):name dCube
+depthcube<float> dCube;
+//TEST_INPUT: Texture2D(size=4, content = zero, arrayLength=2):name d2DArray
+depth2d_array<float> d2DArray;
+//TEST_INPUT: TextureCube(size=4, content = zero, arrayLength=2):name dCubeArray
+depthcube_array<float> dCubeArray;
+
+//TEST_INPUT: Sampler:name samplerState
+SamplerState samplerState;
+//TEST_INPUT: Sampler:name shadowSampler
+SamplerComparisonState shadowSampler;
+
+
+__generic<T:__BuiltinArithmeticType, let N:int>
+bool TEST_texture(
+    Texture1D<vector<T,N>> t1D,
+    Texture2D<vector<T,N>> t2D,
+    Texture3D<vector<T,N>> t3D,
+    TextureCube<vector<T,N>> tCube,
+    Texture1DArray<vector<T,N>> t1DArray,
+    Texture2DArray<vector<T,N>> t2DArray,
+    TextureCubeArray<vector<T,N>> tCubeArray
+)
+{
+    // METAL-LABEL: TEST_texture
```

Comment:
Metal again, and in some other places below.
Why does it say "metal" everywhere?

Reply:
Oh I see it's fixed in a later commit.

Reply:
Sorry about the confusion.
The test file is copied from tests/metal/texture.slang and it was incrementally modified/fixed as I work through the implementation.
The keyword, "Metal", is no longer used in the test file at the latest commit.